modules/system/classes/UpdateManager.php

```diff
@@ -365,21 +365,29 @@ public function uninstall()
     }
 
     /**
-     * Asks the gateway for the lastest build number and stores it.
+     * Asks the gateway (or Composer) for the latest build number and stores it in the database.
      * @return void
      */
     public function setBuildNumberManually()
     {
-        $postData = [];
+        // Use installed.json to detect latest version
+        $composerInstalled = base_path('vendor/composer/installed.json');
+        if (file_exists($composerInstalled)) {
+            $packages = json_decode(file_get_contents($composerInstalled), true);
+            $packageIndex = array_search('october/system', array_column($packages, 'name'));
+            $build = str_replace('v', '', array_get($packages[$packageIndex], 'version', 420));
```

Comment:
@Samuell1 Suggested change to get just the final part of the version as the build:
```suggestion
            $version = str_replace('v', '', array_get($packages[$packageIndex], 'version', '1.0.420'));
            $build = substr($version, strrpos($version, '.') + 1);
```